apps/system/js/accessibility.js

```diff
@@ -284,11 +290,31 @@
     },
 
     /**
+     * Start a timeout that waits to display hints
+     * @memberof Accessibility.prototype
+     */
+    setHintsTimeout: function ar_setHintsTimeout(aDataPlusHints) {
+        clearTimeout(this.hintsTimer);
+        this.hintsTimer = setTimeout(function onHintsTimeout() {
+          if(!this.speechSynthesizer.screen.classList.contains('screenoff')){
```

Comment:
I think a more stable way of doing this is listening to the 'screenchange' event that the screen_manager fires: https://github.com/mozilla-b2g/gaia/blob/master/apps/system/js/screen_manager.js#L556-L559 and checking for it screenEnabled flag in event detail. So you would need to add a new handler for that event in handleEvent. The handler will simply check if screenEnabled is false and if so you would call checkHintsStatus from there.